src/flexalloc_xnvme_env.c

```diff
@@ -189,10 +189,156 @@ fla_xne_sync_seq_w_naddrs(struct xnvme_dev * dev, const uint64_t slba, const uin
   return err;
 }
 
+struct fla_async_strp_cb_args
+{
+  // For the async xfer
+  uint32_t nsid;
+  uint64_t slba;
+  uint16_t nlbs;
+  uint64_t sbuf_nbytes;
+  uint64_t ebuf_nbytes;
+  void * buf;
+  bool write;
+  uint32_t dev_lba_nbytes;
+  uint64_t buf_stride_min_one;
+
+  struct fla_async_cb_args cb_args;
+  struct fla_strp_params *sp;
+};
+
+static void
+fla_async_strp_cb(struct xnvme_cmd_ctx * ctx, void * cb_arg)
```

Comment:
The logic in this function is currently broken. Try a write starting from the second stripe instead of the first. Say we have a strp nobjs = 2 and strp_sz = 32768 bytes